src/main.rs

```diff
@@ -37,15 +37,6 @@ mod usbg_9pfs;
 
 type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
 
-pub fn mkdir(dir: &str) -> Result<()> {
-    if let Err(e) = create_dir(dir) {
-        if e.kind() != io::ErrorKind::AlreadyExists {
-            return Err(format!("Failed to create {dir}: {e}",).into());
```

Comment:
Same here: we loose, which dir could not be created.

And that's not theoretical: the mkdirs in `usbg_9pfs.rs` can fail when drivers are missing in the Kernel.

Reply:
You're right, I didn't think of that. I moved the functions into a `util` module and applied some refactoring.